nixos/modules/config/xdg/portal.nix

```diff
@@ -62,9 +62,12 @@ with lib;
       services.dbus.packages  = packages;
       systemd.packages = packages;
 
+      systemd.user.services.xdg-desktop-portal.environment = {
+        XDG_DESKTOP_PORTAL_DIR = "${joinedPortals}/share/xdg-desktop-portal/portals";
+      };
+
       environment.variables = {
         GTK_USE_PORTAL = mkIf cfg.gtkUsePortal "1";
-        XDG_DESKTOP_PORTAL_DIR = "${joinedPortals}/share/xdg-desktop-portal/portals";
```

Comment:
this should work for the systemd user service if you just change it to `environment.sessionVariables`.

Reply:
Thanks, I'll try that instead!